pulsar-broker/src/main/java/org/apache/pulsar/broker/lookup/TopicLookupBase.java

```diff
@@ -294,9 +295,10 @@ protected String internalGetNamespaceBundle(TopicName topicName) {
                                                 newAuthoritative, LookupType.Redirect, requestId, false));
                             } else {
                                 // When running in standalone mode we want to redirect the client through the service
-                                // url, so that the advertised address configuration is not relevant anymore.
-                                boolean redirectThroughServiceUrl = pulsarService.getConfiguration()
-                                        .isRunningStandalone();
+                                // url, if the advertised address is localhost (per PulsarStandaloneStarter).
+                                ServiceConfiguration conf = pulsarService.getConfiguration();
+                                boolean isLocalhost = StringUtils.equals("localhost", conf.getAdvertisedAddress());
```

Comment:
In standalone mode, what if we set `advertisedAddress` to 127.0.0.1? 
Or the domain name host is set on the machine, but it points to the local.

Reply:
+1

Reply:
Java has an API in order to understand if the address is local, for instance we should support also IPv6 localhost addresses

Reply:
Good idea @eolivelli, I'll use [InetAddress::isLoopbackAddress](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/net/InetAddress.html#isLoopbackAddress()) to test whether the advertised address is a loopback address, in which case we'll force the proxy mode.

Reply:
I updated the code to use `isLoopbackAddress` against the resolved `LookupData`.  The logic can be understood as "don't return a LookupData that is pointing to a loopback address; force a proxy mode in that case".